### PR TITLE
add inventory/product catalogue changes

### PR DESCRIPTION
**Pull request by bennsimon**
1. Adds extra columns for Stocks and StockType

2. Extends similar domains in PlanEvaluator Lib
    - This changes to type of id to Long and  dateCreated to DateTime joda.

3. Adds Download of images belonging to StockType

### CODE REVIEW

File: opensrp-stock/src/main/java/org/smartregister/stock/domain/StockResponse.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.smartregister.stock.domain;
+
+import java.io.Serializable;
+import java.util.List;
+
+public class StockResponse implements Serializable {
```
Comment (githengi) [inline opensrp-stock/src/main/java/org/smartregister/stock/domain/StockResponse.java:6]: Why do we need this class

Comment (bennsimon) [reply opensrp-stock/src/main/java/org/smartregister/stock/domain/StockResponse.java:6]: I used it together with `gson` to convert the JsonObject response to a Pojo

File: opensrp-stock/src/main/java/org/smartregister/stock/configuration/StockSyncConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return false;
+    }
+
+    public StockSyncServiceHelper getStockSyncIntentServiceHelper(){
+        return new StockSyncServiceHelper(this);
+    }
+
+    public boolean useDefaultStockExistenceCheck() {
+        return true;
+    }
```
Comment (qiarie) [inline opensrp-stock/src/main/java/org/smartregister/stock/configuration/StockSyncConfiguration.java:30]: Add method Javadoc comments to better document the configuration options

**Review by qiarie — CHANGES_REQUESTED** [review]

File: opensrp-stock/src/main/java/org/smartregister/stock/StockLibrary.java [hunk 3]
```diff
@@ -36,10 +45,12 @@ public static StockLibrary getInstance() {
         return instance;
     }
 
-    private StockLibrary(Context context, Repository repository, StockExternalRepository stockExternalRepository) {
+    private StockLibrary(Context context, Repository repository, StockExternalRepository stockExternalRepository, AppExecutors appExecutors, StockSyncConfiguration stockSyncConfiguration) {
```
[the remark this answers is not in the document]
Comment (bennsimon) [reply opensrp-stock/src/main/java/org/smartregister/stock/StockLibrary.java:48]: no need .. its method overload

**Review by ndegwamartin — APPROVED** [review]

**Comment by ndegwamartin** [comment]
@bennsimon you need to bump up the artifact's library for publishing

**Pull request merged**